src/legacy.jl

```diff
@@ -0,0 +1,135 @@
+    175 => 0xd787af, 176 => 0xd787d7, 177 => 0xd787ff, 178 => 0xd7af00, 179 => 0xd7af5f,
+    180 => 0xd7af87, 181 => 0xd7afaf, 182 => 0xd7afd7, 183 => 0xd7afff, 184 => 0xd7d700,
+    185 => 0xd7d75f, 186 => 0xd7d787, 187 => 0xd7d7af, 188 => 0xd7d7d7, 189 => 0xd7d7ff,
+    190 => 0xd7ff00, 191 => 0xd7ff5f, 192 => 0xd7ff87, 193 => 0xd7ffaf, 194 => 0xd7ffd7,
+    195 => 0xd7ffff, 196 => 0xff0000, 197 => 0xff005f, 198 => 0xff0087, 199 => 0xff00af,
+    200 => 0xff00d7, 201 => 0xff00ff, 202 => 0xff5f00, 203 => 0xff5f5f, 204 => 0xff5f87,
+    205 => 0xff5faf, 206 => 0xff5fd7, 207 => 0xff5fff, 208 => 0xff8700, 209 => 0xff875f,
+    210 => 0xff8787, 211 => 0xff87af, 212 => 0xff87d7, 213 => 0xff87ff, 214 => 0xffaf00,
+    215 => 0xffaf5f, 216 => 0xffaf87, 217 => 0xffafaf, 218 => 0xffafd7, 219 => 0xffafff,
+    220 => 0xffd700, 221 => 0xffd75f, 222 => 0xffd787, 223 => 0xffd7af, 224 => 0xffd7d7,
+    225 => 0xffd7ff, 226 => 0xffff00, 227 => 0xffff5f, 228 => 0xffff87, 229 => 0xffffaf,
+    230 => 0xffffd7, 231 => 0xffffff, 232 => 0x080808, 233 => 0x121212, 234 => 0x1c1c1c,
+    235 => 0x262626, 236 => 0x303030, 237 => 0x3a3a3a, 238 => 0x444444, 239 => 0x4e4e4e,
+    240 => 0x585858, 241 => 0x626262, 242 => 0x6c6c6c, 243 => 0x767676, 244 => 0x808080,
+    245 => 0x8a8a8a, 246 => 0x949494, 247 => 0x9e9e9e, 248 => 0xa8a8a8, 249 => 0xb2b2b2,
+    250 => 0xbcbcbc, 251 => 0xc6c6c6, 252 => 0xd0d0d0, 253 => 0xdadada, 254 => 0xe4e4e4,
+    255 => 0xeeeeee)
+
+"""
+A mapping from old named colours to the new names, specifically from `light_*`
+to `bright_*`.
+"""
+const RENAMED_COLORS = Dict{Symbol, Symbol}(
+    :light_black => :bright_black,
+    :light_red => :bright_red,
+    :light_green => :bright_green,
+    :light_yellow => :bright_yellow,
+    :light_blue => :bright_blue,
+    :light_magenta => :bright_magenta,
+    :light_cyan => :bright_cyan,
+    :light_white => :bright_white)
+
+legacy_color(color::Symbol) = SimpleColor(get(RENAMED_COLORS, color, color))
+
+function legacy_color(color256::Int)
+    if 0 <= color256 <= 256
+        SimpleColor(ANSI_256_COLORS[color256])
+    else
+        nothing
+    end
+end
+
+"""
+    legacy_color(color::Union{String, Symbol, Int})
+
+Attempt to obtain a `SimpleColor` for a "legacy" color value `color`.
+
+When this is not possible, `nothing` is returned.
+"""
+function legacy_color(color::String)
+    namedcolours = ("black", "red", "green", "yellow", "blue", "magenta", "cyan", "white",
+                    "bright_black", "grey", "gray", "bright_red", "bright_green", "bright_yellow",
+                    "bright_blue", "bright_magenta", "bright_cyan", "bright_white",
+                    "light_black", "light_red", "light_green", "light_yellow",
+                    "light_blue", "light_magenta", "light_cyan", "light_white")
+    if color in namedcolours
+        legacy_color(Symbol(namedcolours))
+    elseif 0 <= (color256 = something(tryparse(Int, ecolor), -1)) <= 255
```

Comment:
Is `ecolor` a typo here?

Reply:
Yep, or more accurately a copy-paste artifact :sweat_smile:.